src/client/components/pages/entities/author.js

```diff
@@ -109,6 +109,38 @@ AuthorAttributes.propTypes = {
 
 function AuthorDisplayPage({entity, identifierTypes, user}) {
 	const urlPrefix = getEntityUrl(entity);
+	const [isSubscribed, setIsSubscribed] = useState(false);
+	useEffect(() => {
+		request.get(`/subscription/entity/isSubscribed/${entity.bbid}`).then(response => {
+			if (response.body.isSubscribed) {
+				setIsSubscribed(true);
+			}
+		});
+	});
+	function handleUnsubscribe(bbid) {
+		const submissionUrl = '/subscription/unsubscribe/entity';
+		request.post(submissionUrl)
+			.send({bbid})
+			.then((res) => {
+				setIsSubscribed(false);
+			}, (error) => {
+				// eslint-disable-next-line no-console
+				console.log('error thrown');
+			});
+	}
+	function handleSubscribe(bbid) {
+		const submissionUrl = '/subscription/subscribe/entity';
```

Comment:
I think this URL structure would be more appropriate and "RESTy":
`/${entityType}/${bbid}/subscribe`  and associated `/unsubscribe` and `/subscribed`
I guess that means we don't need to send any information with that request anymore: BBID is in the route and the user needs to be logged in to access the route so we have their info in the route handler.

That would also work for collections: `/collection/${collectionId}/subscribe`…

That means a fair amount of refactoring on the routes, I know, sorry about that :/
I think you'll end up with a middleware or two (one for entities, one for collections) which should be reusable for each entity's routes. In theory, in these routes you should have access to the user information